src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -23,38 +35,148 @@
+    cargo_motorController = new MotorController("Shooter Cargo", Constants.Shooter.shooterCargoID,40,true);
+    kCargoController = cargo_motorController.getPID();
+    kCargoEncoder = cargo_motorController.getEncoder();
+    shooter_motorController = new MotorController("Shooter", Constants.Shooter.shooterID, 40, true);
     KShooterController = shooter_motorController.getPID();
     KShooterEncoder = shooter_motorController.getEncoder();
-    KShooterController.setP(5e-4);
-    KShooterController.setI(6e-7);
-    KShooterController.setD(0.0);
 
-    KShooterController.setOutputRange(0, 1);
-    /*
-    hood_motorController = new MotorController("Hood", Constants.hoodID);
-    KHoodController = hood_motorController.getPID();
-    KHoodEncoder = shooter_motorController.getEncoder();*/
+    //KShooterController.setP(5e-4);
+    //KShooterController.setI(6e-7);
+    KShooterController.setIMaxAccum(0.9, 0);
+    //KShooterController.setD(0.0);
+
+    //hood_motorController = new MotorController("Hood", Constants.Shooter.hoodID,40,true);
+    //KHoodController = hood_motorController.getPID();
+    //KHoodEncoder = shooter_motorController.getEncoder();
 
   }
+  public void updatePID(){
+
+    
+    if (dashTunePid.getBoolean(false)){
+      
+      Pconstant = PID_P.getDouble(0);
+      Iconstant = PID_I.getDouble(0);
+      Dconstant = PID_D.getDouble(0);
+      Fconstant = PID_F.getDouble(0);
+      MaxOutputConstant = PID_MaxOutput.getDouble(0);
+      MinOutputConstant = PID_MinOutput.getDouble(0);
+      IMaxAccumconstant = PID_IMaxAccum.getDouble(0);
+      IMaxAccumIDconstant =(int)PID_IMaxAccumID.getDouble(0);
+      shooter_motorController.getPID().setOutputRange(MaxOutputConstant, MinOutputConstant);
+      shooter_motorController.getPID().setP(Pconstant);
+      shooter_motorController.getPID().setI(Iconstant);
+      shooter_motorController.getPID().setD(Dconstant);
+      shooter_motorController.getPID().setFF(Fconstant);
+      shooter_motorController.getPID().setIMaxAccum(IMaxAccumconstant, IMaxAccumIDconstant);
+    }
+  }
 
   public void adjustHood(double a) {
     KHoodController.setReference(a, CANSparkMax.ControlType.kPosition);
 
     // Adjusts Hood using PID control to passed angle a
   }
+  public void resetidelay(){
+    idelay = 0;
+  }
+  public double getVelocityInput(){
+    return DShooterRPMInput.getDouble(0.0);
+  }
 
   public void windFlywheel(double rpm) {
+
     // Winds Flywheel using PID control to passed rpm
     // double adjustedRPM = rpm * (Constants.kGearRatioIn / Constants.kGearRatioOut); TODO: reconsider using this
-    currentRPM = rpm;
+    if(rpm ==0.0){
+      KShooterController.setReference(0.0, CANSparkMax.ControlType.kVoltage);
+    }
+    else
+    {
+    rpm = DShooterRPMInput.getDouble(0);
     KShooterController.setReference(rpm, CANSparkMax.ControlType.kVelocity);
+    }
+
+    if (wheelReady()){
+      BCargoRunning.setBoolean(true);
```

Comment:
25 is more efficient

Reply:
the cargo isn't running until your idelay is finished. Don't update to true until it is actually running

Reply:
If wheelReady changes to false while idelay is >0, then it will not run cargo.  

Is this your intention?  Feels like it should be if wheelReady or idelay > 0

Reply:
and why is idelay reset in another method?  Shouldn't it be reset  after line 207?

We should examine whether this part of the logic should be in a command.  Need to look at it further.